reana-ui/src/pages/workflowList/components/WorkflowList.js

```diff
@@ -0,0 +1,209 @@
+        const info = wf.name.split(".");
+        wf.name = info.shift();
+        wf.run = info.join(".");
+        const progress = wf.progress.finished;
+        const total = wf.progress.total;
+        wf.completed = typeof progress === "object" ? progress.total : 0;
+        wf.total = total.total;
+        wf = parseDates(wf);
+      });
+
+      return workflows;
+    }
+    getWorkflows();
+
+    if (!interval.current) {
+      interval.current = setInterval(() => {
+        getWorkflows();
+      }, config.poolingSecs * 1000);
+    }
+
+    return function cleanup() {
+      clearInterval(interval.current);
+    };
+  }, []);
+
+  function parseDates(wf) {
+    const createdMoment = moment.utc(wf.created);
+    const startedMoment = moment.utc(wf.progress.run_started_at);
+    const finishedMoment = moment.utc(wf.progress.run_finished_at);
+    wf.createdDate = createdMoment.format("Do MMM YYYY HH:mm");
+    wf.startedDate = startedMoment.format("Do MMM YYYY HH:mm");
+    wf.finishedDate = finishedMoment.format("Do MMM YYYY HH:mm");
+    wf.friendlyCreated = moment
+      .duration(-moment().diff(createdMoment))
+      .humanize(true);
+    let duration;
+    if (startedMoment.isValid()) {
+      wf.friendlyStarted = moment
+        .duration(-moment().diff(startedMoment))
+        .humanize(true);
+      if (finishedMoment.isValid()) {
+        duration = finishedMoment.diff(startedMoment);
+        wf.friendlyFinished = moment
+          .duration(-moment().diff(finishedMoment))
+          .humanize(true);
+      } else if (startedMoment.isValid()) {
+        duration = moment().diff(startedMoment);
+      }
+
+      const durationMoment = moment.duration(duration);
+      let format;
+      if (durationMoment.hours()) {
+        format = "H[h] m[m] s[s]";
+      } else if (durationMoment.minutes()) {
+        format = "m [min] s [sec]";
+      } else {
+        format = "s [seconds]";
+      }
+      wf.duration = moment.utc(duration).format(format);
+    }
+    return wf;
+  }
+
+  const statusMapping = {
+    finished: { icon: "check circle", color: "green", preposition: "in" },
+    running: { icon: "spinner", color: "blue", preposition: "for" },
+    failed: { icon: "delete", color: "red", preposition: "in" },
+    created: { icon: "file outline", color: "violet" },
+    stopped: {
+      icon: "pause circle outline",
+      color: "yellow",
+      preposition: "after"
+    },
+    queued: { icon: "hourglass outline", color: "teal", preposition: "for" },
+    deleted: { icon: "eraser", color: "gray", preposition: "after" }
+  };
+
+  if (loading) {
+    return (
+      <Dimmer active>
+        <Loader>Loading workflows</Loader>
+      </Dimmer>
+    );
+  } else if (workflows && !workflows.length) {
+    return <NoWorkflows />;
+  } else {
+    return (
+      <Container text className={styles["container"]}>
+        <Title>Your workflows</Title>
+        {workflows &&
+          workflows.length &&
```

Comment:
Is it necessary to check `workflows && workflows.length` again to iterate on the workflow list? We do it already in the previous if.